src/interactionManager/interactionManager.ts

```diff
@@ -6,11 +6,18 @@ import { TransportAPI } from '../types'
 import { SDKError, ErrorCode } from '../errors'
 import { FlowType } from './types'
 
+import { Emitter } from '../events'
+
 const firstMessageForFlowType = {}
-flows.forEach(f => {
+flows.forEach((f) => {
   firstMessageForFlowType[f.type] = f.firstMessageType
 })
 
+export interface InteractionEvents {
+  interactionCreated: (interxn: Interaction) => void
+  interactionUpdated: (interxn: Interaction) => void
+}
```

Comment:
can you export keys which represent event names from here, this way we will avoid some duplication between wallet and sdk repos

Reply:
I guess an enum of available events will be helpful

Reply:
These event names are already enforced by the type checker and if your IDE is clever enough it will autocomplete them even :smile: 

`interactionManager.on('` would already give you the options

How would the enum be used? something like `InteractionEvents.interactionCreated` would just be even longer and even more duplication....

Reply:
I have found what you mean by the enum and have addressed that in my review here: https://github.com/jolocom/smart-wallet-mirror-private/pull/434#discussion_r643227699

Basically use `keyof InteractionEvents` as the type, instead of creating an enum